src/overlays/actors/ovl_En_Dragon/z_en_dragon.c

```diff
@@ -37,130 +72,777 @@ const ActorInit En_Dragon_InitVars = {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->grabTimer == 0) {
+        if (!this->hasGrabbedPlayer) {
+            Math_Vec3f_Copy(&player->actor.world.pos, &this->playerGrabPosition);
+            Math_Vec3f_Copy(&this->playerGrabPositionTemp, &this->playerGrabPosition);
+            this->hasGrabbedPlayer = true;
+        } else {
+            Math_Vec3f_Copy(&player->actor.world.pos, &this->playerGrabPositionTemp);
+        }
+
+        Math_Vec3f_Copy(&sp50, &this->burrowEntrancePos);
+        sp50.x += Math_SinS(this->actor.world.rot.y) * -930.0f;
+        sp50.z += Math_CosS(this->actor.world.rot.y) * -930.0f;
+        Math_Vec3f_Copy(&this->actor.world.pos, &sp50);
+        this->jawZRotation = 0x1450;
+        this->actor.speedXZ = 60.0f;
+    }
+
+    this->grabTimer++;
+    Math_SmoothStepToS(&this->actor.shape.rot.z, zRotPerPython[this->pythonIndex], 0xA, 0x1F4, 0x14);
+    EnDragon_SpawnBubbles(this, globalCtx, this->jawPos);
+
+    Math_Vec3f_Copy(&sp50, &this->burrowEntrancePos);
+    sp50.x += Math_SinS(this->actor.world.rot.y) * cameraEyePerPython[this->pythonIndex].x;
+    sp50.y += cameraEyePerPython[this->pythonIndex].y;
+    sp50.z += Math_CosS(this->actor.world.rot.y) * cameraEyePerPython[this->pythonIndex].z;
+
+    Math_Vec3f_Copy(&at, &this->actor.world.pos);
+    at.x += Math_SinS(this->actor.world.rot.y) * cameraAtPerPython[this->pythonIndex].x;
+    at.y += cameraAtPerPython[this->pythonIndex].y;
+    at.z += Math_CosS(this->actor.world.rot.y) * cameraAtPerPython[this->pythonIndex].z;
+
+    EnDragon_CameraSetAtEye(this, globalCtx, sp50, at);
+
+    if (this->grabTimer > maxGrabTimerPerPython[this->pythonIndex]) {
+        if (this->state == DEEP_PYTHON_GRAB_STATE_START) {
+            func_800B7298(globalCtx, &this->actor, 6);
+            this->state = DEEP_PYTHON_GRAB_STATE_GRABBED;
+        }
+
+        globalCtx->unk_18770(globalCtx, player);
+        player->actor.parent = &this->actor;
+        player->unk_AE8 = 50;
+        this->action = DEEP_PYTHON_ACTION_GRAB;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_UTSUBO_EAT);
+        EnDragon_SetupAttack(this);
+    }
+}
+
+void EnDragon_SetupAttack(EnDragon* this) {
+    if (this->animationIndex != DEEP_PYTHON_ANIMATION_LARGE_SIDE_SWAY) {
+        EnDragon_ChangeAnimation(this, DEEP_PYTHON_ANIMATION_LARGE_SIDE_SWAY);
+    }
+
+    this->behindJawRetreatTimer = 0;
+    this->grabTimer = 0;
+    this->state = DEEP_PYTHON_ATTACK_STATE_START;
+    this->actionFunc = EnDragon_Attack;
+}
+
+void EnDragon_Attack(EnDragon* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 currentFrame = this->skelAnime.curFrame;
+    Vec3f sp4C; // used as both the extended position and the camera eye
+    Vec3f at;
+
+    SkelAnime_Update(&this->skelAnime);
+    Math_SmoothStepToS(&this->actor.shape.rot.z, 0, 0xA, 0x1388, 0);
+    if (!(globalCtx->gameplayFrames & 0xF)) {
```

Comment:
I assume this doesn't match, but I'm going to ask anyways
```suggestion
    if ((globalCtx->gameplayFrames % 16) == 0) {
```

Reply:
I am absolutely shocked, but it *does* match.